Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -26,16 +26,117 @@
 #if ENABLE(ENCRYPTED_MEDIA) && USE(OPENCDM)
 
 #include "CDMPrivate.h"
+#include "GenericTaskQueue.h"
 #include "MediaKeyMessageType.h"
 #include "MediaKeysRequirement.h"
 #include "MediaPlayerPrivateGStreamerBase.h"
 
 #include <gst/gst.h>
+#include <wtf/NeverDestroyed.h>
 #include <wtf/text/Base64.h>
 
 GST_DEBUG_CATEGORY_EXTERN(webkit_media_opencdm_decrypt_debug_category);
 #define GST_CAT_DEFAULT webkit_media_opencdm_decrypt_debug_category
 
+namespace {
+
+using OCDMKeyStatus = KeyStatus;
+
+class OCDMSessionCallbacksNotifier {
+    WTF_MAKE_NONCOPYABLE(OCDMSessionCallbacksNotifier);
+public:
+    class Observer {
+    public:
+        virtual ~Observer() { }
+        virtual void challengeGeneratedCallback(String&&) = 0;
+        virtual void keyUpdatedCallback(String&&) = 0;
```

Comment:
Am I wrong to think that we can remove the parameter of `keyUpdatedCallback` because we're always passing `String()` ? I guess this should be removed in the whole onion.

Reply:
It's not always and empty String or this cannot be assumed since it origins from OCDM

Reply:
No I see whwre the confusion comes from: I call these functions explicitly passing an empty string in case an OCDM function returns an error because my assumption is the notification(s) won't be called by OCDM then